Flaky test in `errorstore` (panic)

https://github.com/fleetdm/fleet/actions/runs/15963634706

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x941963]

goroutine 21 [running]:
github.com/fleetdm/fleet/v4/server/errorstore.(*Handler).storeError(0xc0002f4410, {0xc000304ee0?, 0x4ba3d1?}, {0xc77360, 0xc000382018})
```

## QA

The fix modifies error store logic (when we store errors in Redis) to check for a nil condition. We have only seen this issue in tests.

Nothing specific to do for QA, just to be aware of this during smoke tests.

--- a/changes/30455-errorstore-panic
+++ b/changes/30455-errorstore-panic
@@ -1,0 +1,1 @@
+Fixed potential panic in error handler when Redis is down.

--- a/server/errorstore/errors.go
+++ b/server/errorstore/errors.go
@@ -208,6 +208,14 @@ func (h *Handler) storeError(ctx context.Context, err error) {
 		return
 	}
 
+	// Check if pool is nil to prevent panic
+	if h.pool == nil {
+		if h.testOnStore != nil {
+			h.testOnStore(fmt.Errorf("redis pool is nil"))
+		}
+		return
+	}
+
 	// not using a connection that follows redirections here
 	// in order to do pipeline commands
 	conn := h.pool.Get()

--- a/server/errorstore/errors.go
+++ b/server/errorstore/errors.go
@@ -12,6 +12,7 @@ import (
 	"crypto/sha256"
 	"encoding/base64"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
 	"strconv"
@@ -211,7 +212,7 @@ func (h *Handler) storeError(ctx context.Context, err error) {
 	// Check if pool is nil to prevent panic
 	if h.pool == nil {
 		if h.testOnStore != nil {
-			h.testOnStore(fmt.Errorf("redis pool is nil"))
+			h.testOnStore(errors.New("redis pool is nil"))
 		}
 		return
 	}

